Assets/Scripts/Interceptors/InterceptorBase.cs

```diff
@@ -69,19 +71,35 @@ protected override void Start() {
   protected override void FixedUpdate() {
     base.FixedUpdate();
 
-    // Update the planned number of sub-interceptors remaining.
-    // TODO(titan): Update the planned number of sub-interceptors remaining when the number of leaf
-    // hierarchical objects changes, such as when a new target is added.
-    List<IHierarchical> leafHierarchicals =
-        HierarchicalAgent.LeafHierarchicals(activeOnly: false, withTargetOnly: false);
-    NumSubInterceptorsPlannedRemaining = NumSubInterceptors - leafHierarchicals.Count;
-
     // Check whether the interceptor has a target. If not, request a new target from the parent
     // interceptor.
     if (HierarchicalAgent.Target == null || HierarchicalAgent.Target.IsTerminated) {
       OnAssignSubInterceptor?.Invoke(this);
     }
 
+    // Check whether any targets are escaping from the interceptor.
+    if (EscapeDetector != null && HierarchicalAgent.Target != null &&
+        !HierarchicalAgent.Target.IsTerminated) {
+      List<IHierarchical> targetHierarchicals =
+          HierarchicalAgent.Target.LeafHierarchicals(activeOnly: true, withTargetOnly: false);
+      List<IHierarchical> escapingTargets =
+          targetHierarchicals.Where(EscapeDetector.IsEscaping).ToList();
+      foreach (var target in escapingTargets) {
+        OnReassignTarget?.Invoke(target);
+      }
```

Comment:
I cant find where the current interceptor is removed from the target's pursuer list. Shouldn't `target.RemovePursuer(HierarchicalAgent)` be added?

Same in `RegisterMiss`?:
```csharp
foreach (var targetHierarchical in targetHierarchicals) {
  targetHierarchical.RemovePursuer(HierarchicalAgent);
  OnReassignTarget?.Invoke(targetHierarchical);
}
```

Reply:
It's called when the hierarchical's `Target` changes:
```cs
      if (base.Target != null) {
        base.Target.RemovePursuer(this);
        ClearSubHierarchicals();
        SimManager.Instance.DestroyDummyAgent(Agent.TargetModel);
        Agent.TargetModel = null;
      }
```